recirq/contextuality/ghz_game_2d.py

```diff
@@ -0,0 +1,491 @@
+This is the code used for the implementation of the GHZ game detailed in arxiv.2512.02284
+
+The GHZ Game:
+The GHZ game is a non-communication game that can be won against classical
+strategies if the players share an N-qubit GHZ state before playing the game.
+
+The formal rules of the non-communication game are:
+1. There are N players. Each player $j$ is given a classical input bit $x_j \\in \\{0, 1\\}$,
+   such that the sum of all input bits is even ($\\sum_{j=1}^{N} x_j \\pmod 2 = 0$).
+2. To win, the players must output bits $y_j \\in \\{0, 1\\}$ such that the parity of the
+   output bits equals the parity of half the input sum:
+   $$ \\sum_{j=1}^{N} y_j \\pmod 2 = \frac{1}{2} \\sum_{j=1}^{N} x_j \\pmod 2 $$
+
+This module performs the following steps:
+1. State Preparation: Creating the 2D GHZ state (using H and CZ gates).
+2. Challenge Mapping: Assignments of the input bits $x_j$ (where $\\sum x_j$ is even) are
+   mapped to local measurement settings: If $x_j=1$, measure in Y basis;
+   if $x_j=0$, measure in X basis.
+3. Run Experiment: Performing the measurement on the quantum computer with a sampler.
+4. Winning Condition: The winning condition is met if the sum of the measured output
+   bits ($\\sum y_j$) satisfies the parity condition derived from the challenge.
+
+The experiment workflow is divided into two parts:
+- **GHZ2dExperiment:** Creates and executes all necessary circuits
+  (GHZ game trials and Z-stabilizer checks).
+- **GHZ2dResults:** Stores the raw data and calculates the final fidelity metrics -
+    win_mean, x- and z- type fidelity.
+"""
+
+import collections
+import random
+from typing import Any, cast
+
+import cirq
+import networkx as nx
+import numpy as np
+
+
+def _transform_circuit(circuit: cirq.Circuit) -> cirq.Circuit:
+    """Transforms a Cirq circuit by applying a series of modifications.
+    This is an internal helper function used exclusively by
+    `generate_2d_ghz_circuit` when `add_dd_and_align_right` is True.
+
+    The transformations for a circuit include:
+    1. Adding a measurement to all qubits with a key 'm'.
+       It serves as a stopping gate for the DD operation.
+    2. Aligning the circuit and merging single-qubit gates.
+    3. Stratifying the operations based on qubit count
+    (1-qubit and 2-qubit gates).
+    4. Applying dynamical decoupling to mitigate noise.
+    5. Removing the final measurement operation to yield
+       the state preparation circuit.
+
+    Args:
+        circuit: A cirq.Circuit object.
+
+    Returns:
+        The modified cirq.Circuit object.
+    """
+    qubits = list(circuit.all_qubits())
+    circuit = circuit + cirq.Circuit(cirq.M(*qubits, key="m"))
+    circuit = cirq.align_right(cirq.merge_single_qubit_gates_to_phxz(circuit))
+    circuit = cirq.stratified_circuit(
+        circuit[::-1], categories=[lambda op: cirq.num_qubits(op) == k for k in (1, 2)]
+    )[::-1]
+    circuit = cirq.add_dynamical_decoupling(circuit)
```

Comment:
Can you also add a check here that this function exists?  (or surround it with a try/except block).

This function didn't exist in cirq 1.3, which ReCirq still supports.